packages/jsii-pacmak/lib/targets/python.ts

```diff
@@ -0,0 +1,344 @@
+/* tslint:disable */
+import path = require('path');
+import util = require('util');
+
+import * as spec from 'jsii-spec';
+import { Generator, GeneratorOptions } from '../generator';
+import { Target, TargetOptions } from '../target';
+import { CodeMaker } from 'codemaker';
+import { shell } from '../util';
+
+export default class Python extends Target {
+    protected readonly generator = new PythonGenerator();
+
+    constructor(options: TargetOptions) {
+        super(options);
+    }
+
+    public async build(sourceDir: string, outDir: string): Promise<void> {
+        // Format our code to make it easier to read, we do this here instead of trying
+        // to do it in the code generation phase, because attempting to mix style and
+        // function makes the code generation harder to maintain and read, while doing
+        // this here is easy.
+        await shell("black", ["--py36", sourceDir], {});
+
+        return this.copyFiles(sourceDir, outDir);
+    }
+}
+
+// ##################
+// # CODE GENERATOR #
+// ##################
+const debug = function(o: any) {
+    console.log(util.inspect(o, false, null, true));
+}
+
+
+class Module {
+
+    readonly name: string;
+    readonly assembly?: spec.Assembly;
+    readonly assemblyFilename?: string;
+
+    private buffer: object[];
+
+    constructor(ns: string, assembly?: [spec.Assembly, string]) {
+        this.name = ns;
+
+        if (assembly != undefined) {
+            this.assembly = assembly[0];
+            this.assemblyFilename = assembly[1];
+        }
+
+        this.buffer = [];
+    }
+
+    // We're purposely replicating the API of CodeMaker here, because CodeMaker cannot
```

Comment:
It probably would yea, I duplicated it here primarily for expediency, and because I don't feel wholly comfortable submitting code to an OSS JS project since I'm not super familiar with the ecosystem. Although having duplicated it here will make some things easier in general (for instance, when I get support for typing Named types, being able to add imports for the named types, which we won't know what types to emit imports for until we've hit a ``onMethod`` or similar that has used them).